### PR TITLE
nixos/iso-image: add an option to disable BIOS boot

### PR DESCRIPTION
**Pull request by tie**
###### Description of changes

This change adds an option to disable legacy BIOS boot support for ISO images. The implementation uses syslinux package that currently cannot be cross-compiled from non-x86 platforms, e.g. `aarch64-linux` system.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
   - Cross-compiled from `aarch64-linux` and and successfully booted the resulting ISO using Ventoy.
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/installer/cd-dvd/iso-image.nix [hunk 1]
```diff
@@ -535,10 +535,17 @@ in
...
+      description = lib.mdDoc ''
+        Whether the ISO image should be a BIOS-bootable disk.
+      '';
+    };
+
     isoImage.makeEfiBootable = mkOption {
       default = false;
       description = lib.mdDoc ''
-        Whether the ISO image should be an efi-bootable volume.
+        Whether the ISO image should be an EFI-bootable volume.
```
Comment (RaitoBezarius) [inline nixos/modules/installer/cd-dvd/iso-image.nix:548]: This is an unrelated change, please put it in another commit or drop it from this PR (otherwise it would mess the blame)

**Review by RaitoBezarius — APPROVED** [review]
I am not too shocked from this change but I would like more feedback from maintainers on this.

File: nixos/modules/installer/cd-dvd/iso-image.nix (hunk 1 above)
Comment (tie) [reply nixos/modules/installer/cd-dvd/iso-image.nix:548]: Done, this change is now in separate commit.

**Comment by tie** [comment]
@RaitoBezarius @alyssais gentle ping, are there any updates on this?

File: nixos/modules/installer/cd-dvd/iso-image.nix [hunk 2]
```diff
@@ -535,10 +535,17 @@ in
       '';
     };
 
+    isoImage.makeBiosBootable = mkOption {
+      default = false;
```
Comment (amjoseph-nixpkgs) [inline nixos/modules/installer/cd-dvd/iso-image.nix:539]: Why is the default `false`?
